client/my-sites/post-type-list/post-thumbnail.jsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import resizeImageUrl from 'lib/resize-image-url';
+import { getNormalizedPost } from 'state/posts/selectors';
+
+function PostTypeListPostThumbnail( { post } ) {
+	if ( ! post || ! post.canonical_image ) {
+		return null;
+	}
+
+	return (
+		<div className="post-type-list__post-thumbnail-wrapper">
+			<img
+				src={ resizeImageUrl( post.canonical_image.uri, { w: 80 } ) }
+				className="post-type-list__post-thumbnail" />
+		</div>
+	);
+}
+
+PostTypeListPostThumbnail.propTypes = {
+	globalId: PropTypes.string
```

Comment:
>On that note should the prop be changed to a string representing `post.canonical_image.uri` and then this could be a pure component?

Few notes here:

- Any component wrapped by `connect` is already pure by default, unless passing `{ pure: false }` as the fourth argument ([docs](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options))
- It's fine for this to be pure with an object argument. Since state isn't mutated, we can trust shallow equality test on props.
- Typically I'll try to keep the `connect` selector as minimal as possible, because it is run every single time `dispatch` is called in the app
- That said, if thinking of this component in [the presentational / container mindset](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0), the visual component should simply accept the image to be rendered, not the entire post, so probably makes sense to change